### PR TITLE
Add automations resource REST specification to Microsoft.Security

### PR DESCRIPTION
**Pull request by mikreite**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by mikreite** [comment]
dear reviewers, there is a spell check the fails on the new resource name "automations", please advise, thanks!

**Comment by mikreite** [comment]
@njuCZ  , there is a spell check the fails on the new resource name "automations", please advise, thanks!

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+        {
+          "$ref": "#/definitions/AutomationAction"
+        }
+      ],
+      "properties": {
+        "eventHubResourceId": {
+          "type": "string",
+          "description": "The Event Hub Azure Resource ID."
+        },
+        "connectionString": {
```
Comment (anthony-c-martin) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json]: Does this include creds? If so, please confirm you're not going to return it in a GET.

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/automations/{automationName}/validate": {
```
[the remark this answers is not in the document]
Comment (mikreite) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json]: yes it is, however, there is a wizard scenario where a validation should be performed without the actual saving.

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json (hunk 1 above)
Comment (mikreite) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json]: Thanks for noticing it, the connection string will not be returned in GET

Comment (mikreite) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json]: @anthony-c-martin , in a second thought - why can't they be returned back to the user? they are provided by the user. Appreciate more info and what should be the best practice here

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/Automations/GetAutomationResourceGroup_example.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                  ]
+                }
+              ]
+            }
+          ],
+          "actions": [
+            {
+              "logicAppResourceId": "/subscriptions/e54a4a18-5b94-4f90-9471-bd3decad8a2e/resourceGroups/sample/providers/Microsoft.Logic/workflows/MyTest1",
+              "actionType": "LogicApp",
+              "uri": "(This value contains a secret and will not be retrieved)"
```
[the remark this answers is not in the document]
Comment (mikreite) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/Automations/GetAutomationResourceGroup_example.json]: same question I asked above, regarding your comment on "connectionString" property

[the remark this answers is not in the document]
Comment (mikreite) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/Automations/GetAutomationResourceGroup_example.json]: right, will be handled on our next development iteration

**Comment by mikreite** [comment]
@anthony-c-martin , there is a spell check that fails on the new resource name "automations", please advise how it should be white-listed

**Comment by anthony-c-martin** [comment]
> @anthony-c-martin , there is a spell check that fails on the new resource name "automations", please advise how it should be white-listed

ARM doesn't own these validations. @njuCZ please could you answer @mikreite's question?

**Comment by njuCZ** [comment]
@mikreite sorry for not noticing this thread. How is the matter now? It seems all CI/CD have passed, and there are some new changes after `ARMSignedOff`? So I guess it needs to be reviewed by ARM team again ?

**Comment by mikreite** [comment]
hi @njuCZ , the important changes were required by ARM and reviewed already, no further review is needed

**Comment by njuCZ** [comment]
@mikreite LGTM, I will merge this PR

**Review by njuCZ — APPROVED** [review]

**Comment by njuCZ** [comment]
It seems I can not merge this MR because of there is a requested changes, @mikreite could you please resolve it

**Comment by mikreite** [comment]
Hi Chen and Anthony,

It seems that you are required to approve the PR.
Could you please approve?

Thanks,
Miri

**Review by anthony-c-martin — APPROVED** [review]

**Pull request merged**